### PR TITLE
Own shop hw5

### PR DESCRIPTION
**Pull request by Vosker**

### CODE REVIEW

File: src/main/webapp/WEB-INF/views/orders/info.jsp [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+                <c:out value="${product.price}"/>
+            </td>
+        </tr>
+    </c:forEach>
+</table>
+<br/>
+<a href="${pageContext.request.contextPath}/">To the main page</a>
+<br/>
+</body>
+</html>
```
Comment (SetMyDream) [inline src/main/webapp/WEB-INF/views/orders/info.jsp]: +empty lines =)

**Review by SetMyDream — APPROVED** [review]
Somewhere no empty lines in the end of file.
Looks fine.

File: src/main/webapp/WEB-INF/views/shopping-cart/products.jsp [hunk 2]
```diff
@@ -11,25 +11,35 @@
...
-            </td>
-        </tr>
-    </c:forEach>
+    <c:set var="userID" value="${userId}" />
+    <c:set var="index" value="0" />
+        <c:forEach var="product" items="${products}">
+            <tr>
+                <td>
+                    <c:out value="${index = index + 1}"/>
+                </td>
```
Comment (SetMyDream) [inline src/main/webapp/WEB-INF/views/shopping-cart/products.jsp:19]: Прикольная тема)
Типа не Id, а номер товара в листе, если я правильно понял..., но тогда в строке 10 может имеет смысл переименовать на №?

Comment (Vosker) [reply src/main/webapp/WEB-INF/views/shopping-cart/products.jsp:19]: Угу

File: src/main/java/com/internet/shop/controllers/carts/RemoveProductFromCartController.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final ShoppingCartService shoppingCartService =
+            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+    private final ProductService productService =
+            (ProductService) injector.getInstance(ProductService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String productId = req.getParameter("index");
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/carts/RemoveProductFromCartController.java]:
```suggestion
        String productId = req.getParameter("id");
```
index and id are different things

File: src/main/java/com/internet/shop/controllers/carts/RemoveProductFromCartController.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        int id = Integer.parseInt(productId);
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/carts/RemoveProductFromCartController.java]: use Long please

File: src/main/java/com/internet/shop/controllers/carts/RemoveProductFromCartController.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        shoppingCartService.get(USER_ID).getProducts().remove(id);
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/carts/RemoveProductFromCartController.java]: do we have a special method in `shoppingCartService`?

File: src/main/java/com/internet/shop/controllers/carts/ShowCartController.java [hunk 6]
```diff
@@ -21,6 +21,7 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
         List<Product> allProducts = shoppingCartService.getByUserId(USER_ID).getProducts();
         req.setAttribute("products", allProducts);
+        req.setAttribute("userId", USER_ID);
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/carts/ShowCartController.java]: do we want to show this in JSP? I think we should not display this information to the user

File: src/main/java/com/internet/shop/controllers/orders/AddOrderController.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+import com.internet.shop.lib.Injector;
+import com.internet.shop.service.OrderService;
+import com.internet.shop.service.ShoppingCartService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AddOrderController extends HttpServlet {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/orders/AddOrderController.java]:
```suggestion
public class CompleteOrderController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/orders/AddOrderController.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    private final ShoppingCartService shoppingCartService =
+            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+    private final OrderService orderService =
+            (OrderService) injector.getInstance(OrderService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/orders/all.jsp").forward(req, resp);
+    }
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/orders/AddOrderController.java]: where do you use this?

File: src/main/java/com/internet/shop/controllers/orders/AdminAllOrdersController.java [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+import com.internet.shop.model.Order;
+import com.internet.shop.service.OrderService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AdminAllOrdersController extends HttpServlet {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/orders/AdminAllOrdersController.java]:
```suggestion
public class ShowAllOrdersController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/orders/DeleteOrderController.java [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+
+public class DeleteOrderController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final OrderService orderService =
+            (OrderService) injector.getInstance(OrderService.class);
+
+    @Override
+    public void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String userId = req.getParameter("id");
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/orders/DeleteOrderController.java]: userId or orderId?

File: src/main/java/com/internet/shop/controllers/orders/GetUserOrdersController.java [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+import com.internet.shop.model.Order;
+import com.internet.shop.service.OrderService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class GetUserOrdersController extends HttpServlet {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/orders/GetUserOrdersController.java]:
```suggestion
public class ShowUserOrdersController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/orders/ShowOrderInfoController.java [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+    private final OrderService orderService =
+            (OrderService) injector.getInstance(OrderService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Long orderId = Long.valueOf(req.getParameter("id"));
+        List<Product> products = orderService.get(orderId).getProducts();
+        req.setAttribute("orderId", orderId);
+        req.setAttribute("products", products);
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/orders/ShowOrderInfoController.java]: can we just set `order` as attribute?

File: src/main/java/com/internet/shop/controllers/products/AdminAllProductsController.java [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+import com.internet.shop.model.Product;
+import com.internet.shop.service.ProductService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AdminAllProductsController extends HttpServlet {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/products/AdminAllProductsController.java]:
```suggestion
public class ManageProductsController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/products/DeleteProductController.java [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+
+public class DeleteProductController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final ProductService productService =
+            (ProductService) injector.getInstance(ProductService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String userId = req.getParameter("id");
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/products/DeleteProductController.java]: userId or productId?

File: src/main/webapp/WEB-INF/views/shopping-cart/products.jsp (hunk 2 above)
Comment (KseniiaMakarova) [reply src/main/webapp/WEB-INF/views/shopping-cart/products.jsp:19]: Agree, let's name this column `№`

File: src/main/webapp/WEB-INF/views/shopping-cart/products.jsp [hunk 15]
```diff
@@ -11,25 +11,35 @@
...
+                <td>
+                    <c:out value="${product.name}"/>
+                </td>
+                <td>
+                    <c:out value="${product.price}"/>
+                </td>
+                <td>
+                    <a href="${pageContext.request.contextPath}
+                    /shopping-cart/remove?index=${index - 1}">Remove product</a>
+                </td>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/shopping-cart/products.jsp]: we should pass product actual id here

File: src/main/webapp/WEB-INF/views/orders/info.jsp [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>Order</title>
+</head>
+<body>
+<h1>Order ${orderId}</h1>
+<table border="1">
+    <tr>
+        <th>ID</th>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/orders/info.jsp]: same here - you display a number - not the id

File: src/main/webapp/web.xml [hunk 17]
```diff
@@ -14,14 +14,74 @@
...
     </servlet-mapping>
 
+    <servlet>
+        <servlet-name>addOrder</servlet-name>
+        <servlet-class>com.internet.shop.controllers.orders.AddOrderController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>addOrder</servlet-name>
+        <url-pattern>/orders/add</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/orders/complete</url-pattern>
```

File: src/main/webapp/web.xml [hunk 18]
```diff
@@ -14,14 +14,74 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>allOrders</servlet-name>
+        <servlet-class>com.internet.shop.controllers.orders.GetUserOrdersController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>allOrders</servlet-name>
+        <url-pattern>/orders/all</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/user/orders</url-pattern>
```

File: src/main/webapp/web.xml [hunk 19]
```diff
@@ -14,14 +14,74 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>adminOrders</servlet-name>
+        <servlet-class>com.internet.shop.controllers.orders.AdminAllOrdersController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>adminOrders</servlet-name>
+        <url-pattern>/admin/orders/adminAll</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/admin/orders</url-pattern>
```

File: src/main/webapp/web.xml [hunk 20]
```diff
@@ -34,6 +94,16 @@
...
     </servlet-mapping>
 
+    <servlet>
+        <servlet-name>adminProducts</servlet-name>
+        <servlet-class>com.internet.shop.controllers.products.AdminAllProductsController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>adminProducts</servlet-name>
+        <url-pattern>/admin/products/adminAll</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/admin/products</url-pattern>
```

File: src/main/webapp/web.xml [hunk 21]
```diff
@@ -34,6 +94,16 @@
         <url-pattern>/products/all</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/products</url-pattern>
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/webapp/WEB-INF/views/shopping-cart/products.jsp (hunk 15 above)
Comment (Vosker) [reply src/main/webapp/WEB-INF/views/shopping-cart/products.jsp]: I have a list of products in my shoppingCart. I remove a specific product on a specific position in the list. Is that logic incorrect? If i remove product by id, i need to remove all products with that id, however I may want to delete only a single position.

File: src/main/java/com/internet/shop/controllers/carts/RemoveProductFromCartController.java (hunk 4 above)
Comment (Vosker) [reply src/main/java/com/internet/shop/controllers/carts/RemoveProductFromCartController.java:24]: Глянь плиз мой коммент чуть ниже.

File: src/main/webapp/WEB-INF/views/shopping-cart/products.jsp (hunk 15 above)
Comment (Vosker) [reply src/main/webapp/WEB-INF/views/shopping-cart/products.jsp]: Disregard, we discussed + fixed

File: src/main/java/com/internet/shop/controllers/products/AddProductController.java [hunk 22]
```diff
@@ -17,18 +17,17 @@
...
             throws ServletException, IOException {
-        req.getRequestDispatcher("/WEB-INF/views/products/add.jsp").forward(req, resp);
+        req.getRequestDispatcher("/WEB-INF/views/admin/products/add.jsp").forward(req, resp);
     }
 
     @Override
     protected void doPost(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
-        req.getRequestDispatcher("/WEB-INF/views/products/add.jsp").forward(req, resp);
+        req.getRequestDispatcher("/WEB-INF/views/admin/products/add.jsp").forward(req, resp);
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/products/AddProductController.java]: do we need this line at all? what's its purpose here?

File: src/main/webapp/WEB-INF/views/admin/products/adminAll.jsp [hunk 23]
```diff
@@ -0,0 +1,37 @@
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>Admin</title>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/admin/products/adminAll.jsp]:
```suggestion
    <title>All products</title>
```

let's make titles informative

File: src/main/webapp/WEB-INF/views/shopping-cart/products.jsp [hunk 24]
```diff
@@ -7,28 +7,38 @@
...
-            </td>
-            <td>
-                <c:out value="${product.name}"/>
-            </td>
-            <td>
-                <c:out value="${product.price}"/>
-            </td>
-        </tr>
-    </c:forEach>
+    <c:set var="userID" value="${userId}" />
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/shopping-cart/products.jsp]: I think we should not display such info to the user

File: src/main/webapp/web.xml [hunk 25]
```diff
@@ -14,14 +14,74 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>deleteOrder</servlet-name>
+        <servlet-class>com.internet.shop.controllers.orders.DeleteOrderController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>deleteOrder</servlet-name>
+        <url-pattern>/admin/orders/delete</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/orders/delete</url-pattern>
```

File: src/main/webapp/web.xml [hunk 26]
```diff
@@ -14,14 +14,74 @@
...
+
     <servlet>
         <servlet-name>addProduct</servlet-name>
         <servlet-class>com.internet.shop.controllers.products.AddProductController</servlet-class>
     </servlet>
 
     <servlet-mapping>
         <servlet-name>addProduct</servlet-name>
-        <url-pattern>/products/add</url-pattern>
+        <url-pattern>/admin/products/add</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/products/add</url-pattern>
```

File: src/main/webapp/web.xml [hunk 27]
```diff
@@ -14,14 +14,74 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>deleteProduct</servlet-name>
+        <servlet-class>com.internet.shop.controllers.products.DeleteProductController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>deleteProduct</servlet-name>
+        <url-pattern>/admin/products/delete</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/products/delete</url-pattern>
```

File: src/main/webapp/web.xml [hunk 28]
```diff
@@ -31,7 +91,17 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>adminProducts</servlet-name>
+        <servlet-class>com.internet.shop.controllers.products.ManageProductsController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>adminProducts</servlet-name>
+        <url-pattern>/admin/products</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/products/manage</url-pattern>
```

File: src/main/webapp/web.xml [hunk 29]
```diff
@@ -44,6 +114,16 @@
...
     </servlet-mapping>
 
+    <servlet>
+        <servlet-name>removeProductFromCart</servlet-name>
+        <servlet-class>com.internet.shop.controllers.carts.RemoveProductFromCartController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>removeProductFromCart</servlet-name>
+        <url-pattern>/shopping-cart/remove</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/shopping-cart/products/remove</url-pattern>
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/internet/shop/controllers/products/AddProductController.java (hunk 22 above)
Comment (Vosker) [reply src/main/java/com/internet/shop/controllers/products/AddProductController.java]: I guess forgot to remove after some changes) IDK why I need it) Removed

File: src/main/webapp/WEB-INF/views/shopping-cart/products.jsp (hunk 24 above)
Comment (Vosker) [reply src/main/webapp/WEB-INF/views/shopping-cart/products.jsp]: sor forgot to remove

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**